Doc/requirements.txt

```
@@ -6,8 +6,7 @@
 # Sphinx version is pinned so that new versions that introduce new warnings
 # won't suddenly cause build failures. Updating the version is fine as long
 # as no warnings are raised by doing so.
-# PR #104777: Sphinx 6.2 no longer uses imghdr, removed in Python 3.13.
-sphinx==6.2.1
+sphinx~=7.1.0
```

We could do with Sphinx 7 for this fix: https://github.com/python/cpython/pull/110885#discussion_r1359845086

Let's bump and re-review, now the footer lines are fixed, and we're testing CI with the oldest supported Sphinx.

```suggestion
sphinx~=7.2.0
```